### PR TITLE
#158906908 Toast notifications

### PR DESCRIPTION
**Pull request by bahdcoder**
**What does this PR do?**

Add Toast notifications to the application

**Description of Task to be completed?**

- create Connected(ToastMessage) component for displaying toasts
- mount ToastMessage component in App.tsx component
- add test helpers for testing connected components
- add success color to the color palette
- add styles for ToastMessage component
- add tests for ToastMessage component
- add toast reducer for handling toast messages
- add toasts types and interfaces

**How should this be manually tested?**

- launch application
- open redux dev tools
- dispatch the following event in the dev tools:

```js
{
   type: 'activo-web/DISPLAY_TOAST_MESSAGE',
   toast: {
      message: 'bahdcoder',
      type: 'success'
    }
}
```

**Any background context you want to provide?**

xxx

**What are the relevant pivotal tracker stories?**

[158906908](https://www.pivotaltracker.com/story/show/158906908)

**Screenshots (if appropriate)**

screen shot 2018-07-09 at 6 45 10 pm
screen shot 2018-07-09 at 6 45 17 pm

**Questions:**

xxx

### CODE REVIEW

**Comment by bahdcoder** [comment]
@andela-kakpobome @bourgeois247 @Billmike @missating @kevinejiro @Dammyy @sccofield please I have one small problem with this PR. The `x` close button is not vertically aligned. I tried everything. Please, can someone clone, review PR, and while doing that figure it out, and help a brother? Thanks a lot. all stylings are in `ToastMessages.scss`.

**Comment by bahdcoder** [comment]
Bug with `x` fixed. thanks @missating .

File: src/components/ToastMessage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    this.state = {
+      toast: null,
+    };
+  }
+  componentWillReceiveProps({ toast }) {
+    this.setState({
+      toast,
+    });
+
+    setTimeout(this.hideToastMessage, 2000);
```
Comment (sccofield) [inline src/components/ToastMessage/index.tsx]: @bahdcoder can we make this timeout an optional parameter and pass it as props? I am envisaging a scenario where we would not want the toast to close by itself just to be sure that the user has seen the toast message.

File: src/store/modules/toasts/index.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { DisplayToastMessageAction } from './interfaces';
+import { DISPLAY_TOAST_MESSAGE } from './types';
+
+// action creators
+
+/**
+ * Display toast message action creator.
+ * @param {ToastMessage} toast
```
Comment (Dammyy) [inline src/store/modules/toasts/index.ts]: Add a line space before @ param

File: src/App.tsx [hunk 3]
```diff
@@ -4,11 +4,13 @@ import { withRouter } from 'react-router-dom';
 
 // components
 import NavBar from 'components/NavBar';
+import ToastMessage from 'components/ToastMessage';
 import Routes from './routes';
 
 // tslint:disable-next-line:variable-name
 export const App = ({ location }) => ((
   <React.Fragment>
+      <ToastMessage />
```
Comment (Dammyy) [inline src/App.tsx]: Check the indentation here

File: src/store/modules/toasts/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  toast: {
+    message,
+    type,
+  },
+  type: DISPLAY_TOAST_MESSAGE,
+});
+
+const toastMessageDefaultState = {};
+
+export const reducer = (
```
Comment (Dammyy) [inline src/store/modules/toasts/index.ts]: Documentation is missing for this reducer

File: src/store/modules/toasts/index.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+): DisplayToastMessageAction => ({
+  toast: {
+    message,
+    type,
+  },
+  type: DISPLAY_TOAST_MESSAGE,
+});
+
+const toastMessageDefaultState = {};
+
```
Comment (Dammyy) [inline src/store/modules/toasts/index.ts]: add a comment here, " **// reducer** " indicating this is the reducer

**Comment by Dammyy** [comment]
@bahdcoder Please add a label to this PR

**Comment by Dammyy** [comment]
@bahdcoder please correct your commit message to conform with our style guide  https://github.com/andela/engineering-playbook/tree/master/5.%20Developing/Conventions#commit-message

File: src/components/ToastMessage/index.tsx (hunk 1 above)
Comment (bahdcoder) [reply src/components/ToastMessage/index.tsx]: @sccofield I don't think this is a scenario that we'll face. If we were to do this, we'll add an optional parameter to the toast body. Please @andela-kakpobome could you share your thoughts on this ?

File: src/components/ToastMessage/ToastMessage.scss [hunk 6]
```diff
@@ -0,0 +1,56 @@
+@import '../../assets/scss/variables';
+@import 'sass-rem';
+
+.toast-message {
+  position: absolute;
+  left: 50%;
+  z-index: 99;
+  display: flex;
+  justify-content: space-between;
```
Comment (Billmike) [inline src/components/ToastMessage/ToastMessage.scss]: `justify-content` is a positioning property. It should come before display

File: src/components/ToastMessage/index.tsx (hunk 1 above)
Comment (andela-kakpobome) [reply src/components/ToastMessage/index.tsx]: Definitely a good suggestion, but I'd opt for putting that on hold until we find that we have a need for that.

**Pull request merged**